rfd/0107-desktop-access-performance-research.md

```diff
@@ -91,12 +89,121 @@ sequenceDiagram
+communication step as compared to IronRDP.
+Therefore, to confirm that it really is IronRDP's RemoteFX codec that's making up the majority of the performance improvement, and not the architectural differences, I tried playing the same video in a session using `FreeRDP`, with `FreeRDP` set to have the same RDP settings
+as does `rdp-rs` (bitmaps with default codec) in Teleport's system. Performance is similar to the naked eye to the unwatchable Teleport example above, suggesting that RemoteFX is indeed the critical variable for the visible improvements:
+
+https://drive.google.com/file/d/1XHHIdsSRyUw2lrftjOYrVSXTK-8URuf0/view?usp=sharing
+
+#### Technical Considerations
+
+##### Option 1
+
+Recall that Teleport's Windows desktop architecture currently looks like the following, with the `rdp-rs` RDP client and bitmap-to-PNG translation residing on the Teleport Windows Desktop Service.
+
+```
+                        Teleport Desktop Protocol                         RDP
+               ------------------------------------------        ---------------------
+               |                                        |        |                   |
++----------------------+     +------------------+  +------------------+     +------------------+
+|                      |     |                  |  | (rdp-rs & b-2-p) |     |                  |
+|  (TDP client)        |     |                  |  |    Teleport      |     |                  |
+|  User's Web Browser  ------|  Teleport Proxy -----  Windows Desktop ------|  Windows Desktop |
+|                      |     |                  |  |     Service      |     |                  |
++----------------------+     +------------------+  +------------------+     +------------------+
+```
+
+Meanwhile, IronRDP's default setup looks like this
+
+```
+                                     RDP
+               ------------------------------------------
+               |                                        |
++----------------------+     +------------------+  +------------------+
+|                      |     |                  |  |                  |
+|  (IronRDP)           |     |                  |  |                  |
+|  User's Web Browser  -------  Devolutions     ----  Windows Desktop |
+|                      |     |  Gateway         |  |                  |
++----------------------+     +------------------+  +------------------+
+```
+
+Given that `IronRDP` and `rdp-rs` are both RDP clients written in Rust, the most obvious option is to simply swap one out for the other.
+As ever, this is easier said than done for a variety of reasons including primarily that `IronRDP` is written to connect specifically to [Devolutions Gateway](https://github.com/Devolutions/devolutions-gateway),
```

Comment:
Is it always necessary to use the gateway? They implemented a client in the tauri gui that doesn't seem to use gateway: https://github.com/Devolutions/IronRDP/blob/master/iron-tauri-client/src-tauri/src/main.rs#L228 I might be wrong about this as I did not fully check how everything works under the hood in the IronRDP

Reply:
Ah interesting point, maybe you're right. I'll take a closer look.